packages/playground/blueprints/src/lib/steps/run-sql.ts

```diff
@@ -0,0 +1,77 @@
+import { StepHandler } from '.';
+import { rm } from './rm';
+
+/**
+ * @inheritDoc runSql
+ * @hasRunnableExample
+ * @example
+ *
+ * <code>
+ * {
+ *		"step": "runSql",
+ *		"sql": {
+ *			"resource": "literal",
+ *			"name": "schema.sql",
+ *			"contents": "SELECT * FROM wp_posts"
+ *		},
+ * }
+ * </code>
+ */
+export interface RunSqlStep<ResourceType> {
+	/**
+	 * The step identifier.
+	 */
+	step: 'runSql';
+	/**
+	 * The SQL to run.
+	 */
+	sql: ResourceType;
+}
+
+/**
+ * Run an SQL file.
+ */
+export const runSql: StepHandler<RunSqlStep<File>> = async (
+	playground,
+	{ sql },
+	progress?
+) => {
+	progress?.tracker.setCaption(`Executing SQL File`);
+
+	await playground.writeFile(
+		'/tmp/___schema-backup___.sql',
+		new Uint8Array(await sql.arrayBuffer())
+	);
+
+	const docroot = await playground.documentRoot;
+
+	const runPhp = await playground.run({
+		code: `<?php
+		require_once ${JSON.stringify(docroot)} . '/wp-load.php';
+
+		$handle = fopen('/tmp/___schema-backup___.sql', 'r');
+		$buffer = '';
+
+		global $wpdb;
+
+		while ($bytes = fgets($handle)) {
+			$buffer .= $bytes;
+
+			if (!feof($handle) && substr($buffer, -1, 1) !== "\n") {
+				continue;
+			}
+
+			if (substr($buffer, 0, 2) === '--') {
+				$buffer = '';
+				continue;
+			}
+
+			$wpdb->query($buffer);
```

Comment:
Let's throw a JS error if any query fails and add an option like `stopOnFailure` so developers can change that behavior

Reply:
https://github.com/WordPress/wordpress-playground/pull/745/files#diff-ef7fc55c9e2c228ab627cd90d8372d926a0d8625f5f8fc76baf10cd15ed4f85fR80

Reply:
it appears that I may have missed a comment I thought I had posted, but work had already begun on a step listener and failure-handler in #605

Reply:
Ah thanks for that @dmsnell! Hm, let's remove the `stopOnFailure` option from here for now and then treat it holistically with #605.

Reply:
Removed.